pandas/core/indexes/datetimes.py

```diff
@@ -801,6 +802,8 @@ def _add_delta(self, delta):
 
         if isinstance(delta, (Tick, timedelta, np.timedelta64)):
             new_values = self._add_delta_td(delta)
+        elif isinstance(delta, np.ndarray) and is_timedelta64_dtype(delta):
```

Comment:
this should simply be incorporated into the case right below

Reply:
by below, do I understand you correctly that we can consolidate as:

```python
if isinstance(delta, (Tick, timedelta, np.timedelta64, np.ndarray)):
    new_values = self._add_delta_td(delta)
elif ..:
```

or are you suggesting to wrap `delta` into a `TimedeltaIndex`?

Reply:
yes
but i think we need to check the dtype as well if it's a numpy array (can be separate from this)